lib/Lift/CodeGenerator.cpp

```diff
@@ -706,6 +751,14 @@ void CodeGenerator::translate(optional<uint64_t> RawVirtualAddress) {
       if (not F->isDeclaration() and not F->isIntrinsic())
         F->setLinkage(GlobalValue::InternalLinkage);
 
+  //
+  // Create the libtcg context
+  //
+  // TODO(anjo): Move to Lift.cpp?
+  LibTcgDesc Desc = {};
```

Comment:
What's the use case for `LibTcgDesc`?
We're not using it.

Reply:
This is from a common C pattern to allow for default initialized and named arguments, it's very useful when it comes to configuring something such as the LibTcgContext in this case.

Currently LibTcgDesc only has two fields
```c
typedef struct LibTcgDesc {
    void *(*mem_alloc)(size_t);
    void (*mem_free)(void *);
} LibTcgDesc;
```

for overriding the memory allocator used for translation blocks/temporaries, initializing it to 0, means "use the default" which is malloc/free in this case.

A real-world use-case is in the libtcg example analyzes where we override the allocator to be a simple bump allocator
```c
    LibTcgInterface libtcg;
    LibTcgContext *context;
    libtcg_open(arch, &(LibTcgDesc){
        .mem_alloc = libtcg_alloc,
    }, &libtcg, &context);
```
Here we are overriding the allocator to a simple bump allocator which uses a single allocation and partitions out the memory, we then free all memory at once later on. Note this uses the libtcg-loader API (libtcg_open), which revng doesnt use ATM.

At the time of designing the libtcg API I was expecting needing to add way more configuration options. This pattern is not as useful in our case when we only have two parameters. I see this being more useful if we in the future decide to add more configuration options to initializing the LibTcgContext.

Here's a more complicated example which uses the same pattern along with designated array initializers in C to construct complicated data in readable ways (from https://github.com/floooh/sokol)
```c
    state.pip = sg_make_pipeline(&(sg_pipeline_desc){
        .shader = sg_make_shader(triangle_shader_desc(sg_query_backend())),
        .layout = {
            .attrs = {
                [ATTR_triangle_position].format = SG_VERTEXFORMAT_FLOAT3,
                [ATTR_triangle_color0].format = SG_VERTEXFORMAT_FLOAT4
            }
        },
    });
```

Reply:
Got it.